### PR TITLE
fix incompatible errors when building with bazel 0.26+

### PR DESCRIPTION
**Pull request by lucifer9**
I am not an expert of bazel, just managed to make it work again.

### CODE REVIEW

**Comment by kslr** [comment]
hi @lucifer9 

other please submit new pr.
Is this compatible with the lower version?

**Comment by lucifer9** [comment]
> hi @lucifer9
> 
> other please submit new pr.
> Is this compatible with the lower version of bazel?

Good question 👍 
I just tested from 0.18.0 to 0.27.0 and they all work.

File: transport/internet/http/hub.go [hunk 1]
```diff
@@ -104,9 +107,35 @@ func Listen(ctx context.Context, address net.Address, port net.Port, streamSetti
 
 	config := tls.ConfigFromStreamSettings(streamSettings)
 	if config == nil {
-		return nil, newError("TLS must be enabled for http transport.").AtWarning()
-	}
+		// return nil, newError("TLS must be enabled for http transport.").AtWarning()
```
Comment (kslr) [inline transport/internet/http/hub.go]: H2C has been sent separately, I suggest you re-initiate PR

File: infra/bazel/zip.bzl [hunk 2]
```diff
@@ -47,10 +47,10 @@ def _zip_file(ctx):
         if (s.startswith("/") or s.endswith("/") or
             d.startswith("/") or d.endswith("/")):
             fail("mappings should not begin or end with slash")
-    srcs = depset()
-    srcs += ctx.files.srcs
-    srcs += ctx.files.data
-    srcs += collect_runfiles(ctx.attr.data)
+    srcs = depset(transitive = [depset(ctx.files.srcs),depset(ctx.files.data),depset(collect_runfiles(ctx.attr.data))])
+    # srcs += ctx.files.srcs
```
[the remark this answers is not in the document]
Comment (lucifer9) [reply infra/bazel/zip.bzl]: OK, I'll do that

File: transport/internet/http/hub.go (hunk 1 above)
Comment (lucifer9) [reply transport/internet/http/hub.go]: You are right, I'll re-initiate PR.

**Pull request closed without merging**